### PR TITLE
Manually "merge" 2.2 refactorings forward

### PR DESCRIPTION
**Pull request by etedpet**

### CODE REVIEW

File: common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java [hunk 1]
```diff
@@ -29,9 +29,18 @@
...
+        this.actualFields = getFieldsAvailableInRecord(auditRecord, configuredFields);
+    }
+
+    final static FieldSelector getFieldsAvailableInRecord(AuditRecord auditRecord, FieldSelector configuredFields)
+    {
+        FieldSelector fields = auditRecord.getBatchId().isPresent()
+                               ? configuredFields
+                               : configuredFields.withoutField(Field.BATCH_ID);
+
+        return auditRecord.getClientAddress() != null
```
Comment (eperott) [inline common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java]: Hmm, perhaps we agreed on somethign else when we discussed this earlier.

But I'm thinking that we shouldn't merge this forward. Reason is that if (when) we drop support for 2.2, we will still have this code in all branches for no good reason.

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestWriteReadVersionCurrent.java [hunk 2]
```diff
@@ -102,6 +102,18 @@ public void tryReuseOnRead() throws Exception
         .withMessage("Tried to read from wire with used marshallable");
     }
 
+    @Test
+    public void testGetFieldsAvailableInRecord()
```
Comment (eperott) [inline common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestWriteReadVersionCurrent.java]: Remove client parts if you agree to my comment above.

File: common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java (hunk 1 above)
Comment (etedpet) [reply common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java]: Ok, I'll remove this change...

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestWriteReadVersionCurrent.java (hunk 2 above)
Comment (etedpet) [reply common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestWriteReadVersionCurrent.java]: remove this as well

**Pull request merged**